extensions/wikia/AdEngine/js/slot/bottomLeaderboard.js

```diff
@@ -26,11 +39,35 @@ define('ext.wikia.adEngine.slot.bottomLeaderboard', [
 			if (!pushed && pushPos < scrollPosition) {
 				pushed = true;
 				doc.removeEventListener('scroll', pushSlot);
-				win.adslots2.push(slotName);
-				log(['pushSlot', 'Pushed ' + slotName], 'debug', logGroup);
+
+				if (!collapseOnSameViewport()) {
+					win.adslots2.push(slotName);
```

Comment:
What about refactoring AdEngine queue a bit, so you can pass all needed informations (maybe full AdSlot object)?